controllers/actions.github.com/autoscalinglistener_controller.go

```diff
@@ -306,6 +306,38 @@ func (r *AutoscalingListenerReconciler) cleanupResources(ctx context.Context, au
 		logger.Info("Listener proxy secret is deleted")
 	}
 
+	listenerRoleBinding := new(rbacv1.RoleBinding)
+	err = r.Get(ctx, types.NamespacedName{Namespace: autoscalingListener.Spec.AutoscalingRunnerSetNamespace, Name: scaleSetListenerRoleName(autoscalingListener)}, listenerRoleBinding)
+	switch {
+	case err == nil:
+		if listenerRoleBinding.ObjectMeta.DeletionTimestamp.IsZero() {
+			logger.Info("Deleting the listener role binding")
+			if err := r.Delete(ctx, listenerRoleBinding); err != nil {
+				return false, fmt.Errorf("failed to delete listener role binding: %v", err)
+			}
+		}
+		return false, nil
+	case err != nil && !kerrors.IsNotFound(err):
+		return false, fmt.Errorf("failed to get listener role binding: %v", err)
+	}
+	logger.Info("Listener role binding is deleted")
+
+	listenerRole := new(rbacv1.Role)
+	err = r.Get(ctx, types.NamespacedName{Namespace: autoscalingListener.Spec.AutoscalingRunnerSetNamespace, Name: scaleSetListenerRoleName(autoscalingListener)}, listenerRole)
```

Comment:
Do we need to create / delete the role in the controller? Why not create the role in the charts and the binding is applied by the controller?

Reply:
After some digging, I figured why :smile: 

We can't because the role is assigned to a specific ephemeral runner set by the name. So if AutoscalingRunnerSet is updated, the EphemeralRunnerSet name will change and at that point, we need to update the role: https://github.com/actions/actions-runner-controller/blob/ea2443a410fce4629488b1fde14ed4244d327d03/controllers/actions.github.com/autoscalinglistener_controller.go#L192